File: 485-max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int c = 0, m = 0;
        int n=nums.length;

        for(int i=0; i<n; ++i) {
            if(nums[i] == 0)
                c = 0;
            else {
                c+=1;
                m = m < c ? c : m;
            }
        }
        return m;
    }
}

File: 485-max-consecutive-ones/max-consecutive-ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    let count=0, maximum=0;
    const n = nums.length;
    
    for(let i=0; i<n; ++i) {
        if(nums[i] == 1) {
            count += 1;
            maximum = Math.max(maximum, count);
        }
        else count = 0;
    }
    return maximum;
};